keras/engine/training.py

```diff
@@ -1831,8 +1831,12 @@ def generate_arrays_from_file(path):
                                  str(validation_data))
             val_x, val_y, val_sample_weights = self._standardize_user_data(
                 val_x, val_y, val_sample_weight)
+            val_data = val_x + val_y + val_sample_weights
+            if self.uses_learning_phase and not isinstance(K.learning_phase(),
+                                                           int):
```

Comment:
Probably an oversight, you can add it. Having a unit test for `callback.validation_data` would also be good.